src/ansys/dpf/composites/_composite_model_impl_2023r2.py

```diff
@@ -0,0 +1,589 @@
+from typing import Collection, Dict, List, Optional, Sequence, cast
+from warnings import warn
+
+import ansys.dpf.core as dpf
+from ansys.dpf.core import FieldsContainer, MeshedRegion, Operator, UnitSystem
+from ansys.dpf.core.server_types import BaseServer
+import numpy as np
+from numpy.typing import NDArray
+
+from .composite_scope import CompositeScope
+from .data_sources import (
+    CompositeDataSources,
+    ContinuousFiberCompositesFiles,
+    get_composites_data_sources,
+)
+from .failure_criteria import CombinedFailureCriterion
+from .layup_info import (
+    ElementInfo,
+    LayerProperty,
+    LayupPropertiesProvider,
+    add_layup_info_to_mesh,
+    get_element_info_provider,
+)
+from .layup_info.material_operators import MaterialOperators, get_material_operators
+from .layup_info.material_properties import MaterialProperty, get_constant_property_dict
+from .result_definition import FailureMeasureEnum, ResultDefinition, ResultDefinitionScope
+from .sampling_point_2023r2 import SamplingPoint2023R2
+from .sampling_point_types import SamplingPoint
+from .server_helpers import upload_continuous_fiber_composite_files_to_server
+from .unit_system import UnitSystemProvider, get_unit_system
+
+
+class CompositeInfo:
+    """Contains composite data providers for a composite definition."""
+
+    def __init__(
+        self,
+        data_sources: CompositeDataSources,
+        composite_definition_label: str,
+        streams_provider: dpf.Operator,
+        material_operators: MaterialOperators,
+        unit_system: UnitSystemProvider,
+    ):
+        """Initialize ``CompositeInfo`` class and add enriched mesh with composite information."""
+        mesh_provider = dpf.Operator("MeshProvider")
+        mesh_provider.inputs.data_sources(data_sources.rst)
+        self.mesh = mesh_provider.outputs.mesh()
+
+        self.layup_provider = add_layup_info_to_mesh(
+            mesh=self.mesh,
+            data_sources=data_sources,
+            material_operators=material_operators,
+            unit_system=unit_system,
+            composite_definition_label=composite_definition_label,
+        )
+
+        self.element_info_provider = get_element_info_provider(
+            mesh=self.mesh,
+            stream_provider_or_data_source=streams_provider,
+        )
+        self.layup_properties_provider = LayupPropertiesProvider(
+            layup_provider=self.layup_provider, mesh=self.mesh
+        )
+
+
+class CompositeModelImpl2023R2:
+    """Provides access to the basic composite postprocessing functionality.
+
+    This class supports DPF Server version up to and including 2023R2 (6.1).
+    On initialization, the ``CompositeModel`` class automatically adds composite lay-up
+    information to the meshed regions. It prepares the providers for different lay-up properties
+    so that they can be efficiently evaluated. The composite_files provided are automatically
+    uploaded to the server if needed.
```

Comment:
```suggestion
    This class supports DPF Server version 6.1 (2-23 R2) and earlier.
    On initialization, the ``CompositeModel`` class automatically adds composite lay-up
    information to the meshed regions. It prepares the providers for different lay-up properties
    so that they can be efficiently evaluated. The composite_files provided are automatically
    uploaded to the server if needed.
```